### ISSUE TITLE
Jet Smart Filter - Listing grid - Can't use filter on archive taxinomie

### ISSUE THREAD

**Issue by Bibliothecaire2** [Crocoblock/suggestions#160]
Hi,

I have create a page for smar filter, work fine : https://madatest.com/hotels/
My question is, how to do the same for the TAXINOMIE archive ?
I'm a guide, so i have a lot of destination (Paris / Antananarivo / london / etc...)
A have create a Taxinomie "destination", so i just want to create a elementor template for the taxinomie "destination", but on the "listing grid" there is no option to automatically detect the taxonomy of the archive and make the filter (as the "archive" of elementor pro, exemple : https://madatest.com/destination/antananarivo/)

To summarize, what is needed is:

Resume the "archive" functionality of elementor archive and add it on the list grid
Make" Smart Filter" Filters Take into Account the Listing Grid Rule
Thanks for your help

**Comment by Biblothecaire** [comment]
Hi,

At the and the taxinomie archive work, i found my error.
BUT other probleme, on post query I have :
"You select Use as Archive Template option, so other query parameters will be ignored"

So currently I have no choice, it is a posting by last published. Not what I need
I need it to optimize the display. Example:
Order & Offet = order by last modifed / or random
Exemple : https://madatest.com/destination/antananarivo/

**Comment by caremedchoice** [comment]
How did you fix this.  What im looking to do is to have a smart filter, and the a taxonomy lisitng.  The filter will filter the taxonomy lisitng.  THX

**Comment by Biblothecaire** [comment]
At the end i need to do it manually on each archive page ...
Long, and the day i will want to change the deseign i'm dead !

**Comment by ihaseebali** [comment]
> 
> 
> At the end i need to do it manually on each archive page ...
> Long, and the day i will want to change the deseign i'm dead !

Hi, but i think this is not a good solution.

**Comment by ihaseebali** [comment]
I am also facing the same issue for Taxonomy Archive Templates,
I did the same as @Bibliothecaire2 did.
But now every time when i add new category i need to create a new template and set conditions...
So this is not good solution, need to resolve this bug as soon as possible.

**Comment by ihaseebali** [comment]
@Crocoblock kindly consider and solve this bug in jet smart filters for custom taxonomy archive pages.

**Comment by hossein7r** [comment]
> At the end i need to do it manually on each archive page ...
> Long, and the day i will want to change the deseign i'm dead !

Not a good solution. Very tiresome and also problematic.
I tried that. The issue is if I set the Grid Query to a certain taxonomy, for example product Category, Then If the user remove the taxonomy filter, the grid still shows that certain taxonomy.
There should be a solution to make just one page for all archive pages without hassle.

@Crocoblock Please solve this issue. Thanks.

**Comment by oddpixel** [comment]
Here is a solution to changing the sort order. Just add to your functions.php or a custom plugin. This just edit's the JetEngine meta query filter.

Here I am checking if the archive is in the custom post type archive (projects) and then adding my custom meta query and sort order. **rating_order** sorts by a number based meta field (a 1-5 rating), **title_order** loads any other posts that do not have a rating assigned to them to the end of the query.

```
function ac_order_projects( $query ) {
	if ( is_admin() ) {
		return;
	}
	
	if ( $query['post_type'] == 'project' ) {

		$meta_query_array = array(
			'relation' => 'OR',
			'rating_order' => array(
				'key' => 'rating',
				'compare' => 'EXISTS',
				'type' => 'NUMERIC'
			),
			'title_order' => array(
				'key' => 'rating',
				'compare' => 'NOT EXISTS',
				'type' => 'NUMERIC'
			),
		);

		$query['meta_query'] = $meta_query_array;
		$query['orderby'] = 'rating_order';
		$query['order'] = 'DESC';
	}

	return $query;
}
add_filter('jet-engine/listing/grid/posts-query-args', 'ac_order_projects');
```

**Comment by simplicity17** [comment]
This really needs to be resolved. I have run into this issue numerous times and had to use REDIRECTS as a workaround. Unacceptable. How hard is it to let us combine "Archive Template" with other query options. Its not like its something "new".

**Comment by tizum** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by ICR17** [comment]
+1

**Comment by AndriiPavlenkoZaelab** [comment]
+1
@Crocoblock any updates?

**Comment by MjHead** [comment]
Listing Grid has "use as archive template" option also Current Query query type can be used for this case

**Comment by sara-kariv** [comment]
> Listing Grid has "use as archive template" option also Current Query query type can be used for this case

Why did you closed this issue as completed?
it's still not working

**Comment by aalexandr21** [comment]
@sarit026 

Please, contact [our support team](https://crocoblock.com/help-center) regarding this case and create a ticket including access details for the site where the issue occurred to ensure we can help you.
